options/glibc/include/net/ethernet.h

```diff
@@ -2,11 +2,16 @@
 #define _NET_ETHERNET_H
 
 #include <stdint.h>
+#include <mlibc-config.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+#ifdef __MLIBC_LINUX_OPTION
+#	include <linux/if_ether.h>
+#endif
```

Comment:
This is a kernel header, so I don't know if this should be hidden behind `__MLIBC_LINUX_OPTION`? Maybe it needs to be an ABI header or hidden behind `#ifdef __linux__`, not sure which.

Reply:
This implementation is (de-facto) "mandated" by Linux, hence my choice to choose based on that, though, this should `#define __UAPI_DEF_ETHHDR 1` before the include, really, unless we want to reimplement it (as musl chooses to for instance). While which one of the two is used is mostly bikeshedding (there's no real standard on this), this forward would be useful on non-Linux systems.

Alternatively, we can just bring out the definitions into mlibc. It's only a struct and a bunch of ethtype defines (and this is what musl does).

Reply:
So does this current implementation actually work? You say we should do ``#define __UAPI_DEF_ETHHDR 1`` before the include, so it's unclear to me whether this solution right now is sufficient.

Reply:
yeah, for pedantics pretty much, the header defaults it to 1:

```c
/* SPDX-License-Identifier: GPL-2.0+ WITH Linux-syscall-note */

/* allow libcs like musl to deactivate this, glibc does not implement this. */
#ifndef __UAPI_DEF_ETHHDR
#define __UAPI_DEF_ETHHDR		1
#endif

#if __UAPI_DEF_ETHHDR
struct ethhdr {
	unsigned char	h_dest[ETH_ALEN];	/* destination eth addr	*/
	unsigned char	h_source[ETH_ALEN];	/* source ether addr	*/
	__be16		h_proto;		/* packet type ID field	*/
} __attribute__((packed));
#endif
```

Reply:
How does musl handle this?

Reply:
Glibc can use cheats, as the kernel has hardcoded checks for glibc in the libc compat header. Musl doesn’t have that and has to do it themselves. We can’t compare with glibc unless we either claim we’re glibc (stupid) or get a fix into the Linux kernel (have fun).